lib/push.js

```diff
@@ -0,0 +1,186 @@
+var client = require('./rest'),
+    constant = require('./constants'),
+    logger = require('./logger'),
+    dispatcher = require('./dispatcher'),
+    channels = require('./push/channels');
+
+function timezone() {
```

Comment:
Moment-timezone is not a viable option here. That's only for displaying purposes, whereas this logic needs to detect.